### PR TITLE
[Merged by Bors] - chore(GroupCat/Injective): golf 

### PR DESCRIPTION
**Pull request by alreadydone**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by jjaassoonn — APPROVED** [review]

File: Mathlib/Algebra/Category/GroupCat/Injective.lean [hunk 1]
```diff
@@ -44,379 +44,148 @@ set_option linter.uppercaseLean3 false
...
-          simpa only [ModuleCat.coe_comp, LinearMap.coe_mk, Function.comp_apply] using
-            FunLike.congr_fun eq1 x
-        rw [cancel_mono] at eq2
-        have : ⇑α' = ⇑β' := congrArg _ eq2
-        ext x
-        apply congrFun this _
-      refine' ⟨(Injective.factorThru G F).toAddMonoidHom, _⟩
-      ext x
-      convert FunLike.congr_fun (Injective.comp_factorThru G F) x}
+    CategoryTheory.Injective (⟨A,inferInstance⟩ : AddCommGroupCat) where
```
Comment (jcommelin) [inline Mathlib/Algebra/Category/GroupCat/Injective.lean:47]: In fact, we know that `ModuleCat \Z` and `AddCommGroupCat` are equivalent categories. So this lemma should really follow from a generic lemma that says that equivalences preserve injectivity of objects. Would you mind factoring out that lemma? You could reuse it in the lemma below this one.

Comment (alreadydone) [reply Mathlib/Algebra/Category/GroupCat/Injective.lean]: Great idea, [golfed](https://github.com/leanprover-community/mathlib4/pull/7204/commits/d95b6e47e07761256a5e884289e2495e0cf830c4).

> a generic lemma that says that equivalences preserve injectivity of objects

There are `Adjunction` lemmas directly applicable so I think there's no point adding a `Equivalence` version.

Comment (alreadydone) [reply Mathlib/Algebra/Category/GroupCat/Injective.lean]: I added the Equivalence anyway: https://github.com/leanprover-community/mathlib4/pull/7204/commits/a2bebbfb7e74c9f40d1fc012b2b8cc5387dc0cfe

File: Mathlib/Algebra/Category/GroupCat/Injective.lean [hunk 2]
```diff
@@ -43,384 +44,129 @@ set_option linter.uppercaseLean3 false
...
+  obtain rfl | h0 := eq_or_ne n 0
+  · apply map_zero
+  apply (AddCircle.coe_eq_zero_iff _).mpr ⟨1, _⟩
+  simp [mul_inv_cancel (Nat.cast_ne_zero (R := ℚ).mpr h0)]
+
+variable {a}
+
+/-- The map sending `n • a` to `n / 2` when `a` has infinite order,
+  and to `n / addOrderOf a` otherwise. -/
 @[simps!] noncomputable def toRatCircle : (ℤ ∙ a) →ₗ[ℤ] AddCircle (1 : ℚ) :=
```
Comment (jcommelin) [inline Mathlib/Algebra/Category/GroupCat/Injective.lean:126]: Should this actually be renamed to `toAddCircle`?

**Review by jcommelin — COMMENTED** [review]
Thank you so much for this monster golf!

**Comment by jcommelin** [comment]
bors d+

File: Mathlib/Algebra/Category/GroupCat/Injective.lean (hunk 2 above)
Comment (alreadydone) [reply Mathlib/Algebra/Category/GroupCat/Injective.lean:126]: RatCircle makes sense as it distinguishes Q/Z from the usual circle R/Z. Anyway this declaration is in a namespace not supposed to be used from outside, so the name shouldn't matter much.

**Comment by alreadydone** [comment]
Thanks!

bors r+

**Pull request closed without merging**